apps/omg_child_chain/lib/omg_child_chain/block_queue/balance.ex

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.BlockQueue.Balance do
```

Comment:
Does this need to be a GenServer? Doesn't look like it's doing anything long running, async, nor periodic or need to be singleton...

Reply:
Not necessarily. Any particular alternative in mind?

I want to isolate and decouple from BlockQueue-but also not put too much work in if we’re writing a new one.

Reply:
Thinking of [`Task.start/1`](https://hexdocs.pm/elixir/Task.html#start/1) but I guess it's overoptimization given how this is called once per chch block, so resolving this thread.